src/tests/sync/src/test_utils.rs

```diff
@@ -0,0 +1,155 @@
+    std::env::set_var("PLAYGROUND_PROFILE", "playground-test");
+    std::env::set_var("OBJECT_STORE_URL", "memory-shared");
+    std::env::set_var("OBJECT_STORE_BUCKET", "hummock_001");
+}
+
+pub async fn start_cluster() -> (JoinHandle<()>, std::sync::mpsc::Sender<()>) {
+    wipe_object_store().await;
+    sync::activate_sync_point(
+        "CLUSTER_READY",
+        vec![sync::Action::EmitSignal("SIG_CLUSTER_READY".to_owned())],
+        1,
+    );
+
+    let (tx, rx) = std::sync::mpsc::channel();
+    let join_handle = std::thread::spawn(move || {
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(1)
+            .enable_all()
+            .build()
+            .unwrap();
+        runtime.block_on(async {
+            tokio::spawn(async { playground().await });
+            rx.recv().unwrap();
+        });
+    });
+    // It will find "SIG_CLUSTER_READY" even when it is reached after "SIG_CLUSTER_READY" has been
+    // emitted.
+    wait_now("SIG_CLUSTER_READY".to_owned(), Duration::from_secs(30))
+        .await
+        .unwrap();
+    (join_handle, tx)
+}
+
+pub async fn stop_cluster(join_handle: JoinHandle<()>, shutdown_tx: std::sync::mpsc::Sender<()>) {
+    shutdown_tx.send(()).unwrap();
+    join_handle.join().unwrap();
+    wipe_object_store().await;
+}
+
+pub fn run_slt() -> Output {
+    Command::new("./risedev")
+        .args([
+            "slt",
+            "-d",
+            "dev",
+            "-p",
+            "4566",
+            "src/tests/sync/slt/tpch_snapshot_no_drop.slt",
+        ])
+        .spawn()
+        .unwrap()
+        .wait_with_output()
+        .unwrap()
+}
+
+pub async fn get_object_store_client() -> ObjectStoreImpl {
+    let url = std::env::var("OBJECT_STORE_URL").unwrap();
+    parse_remote_object_store(&url, Arc::new(ObjectStoreMetrics::unused())).await
+}
+
+pub fn get_object_store_bucket() -> String {
+    std::env::var("OBJECT_STORE_BUCKET").unwrap()
+}
+
+async fn wipe_object_store() {
+    // TODO: wipe content of object store
+}
+
+pub async fn get_meta_client() -> MetaClient {
+    let meta_addr = std::env::var("RW_META_ADDR").unwrap();
+    let mut client = MetaClient::new(&meta_addr).await.unwrap();
+    let worker_id = client
+        .register(WorkerType::Generic, &"127.0.0.1:2333".parse().unwrap(), 0)
+        .await
+        .unwrap();
+    client.set_worker_id(worker_id);
+    client
+}
+
+pub async fn wait_now(signal: Signal, timeout: Duration) -> Result<(), sync::Error> {
+    sync::activate_sync_point(
+        "NOW",
+        vec![sync::Action::WaitForSignal(WaitForSignal {
+            signal,
+            clear_signal: true,
+            timeout,
+        })],
+        1,
+    );
+    sync::on_sync_point("NOW").await
+}
+
+pub async fn emit_now(signal: Signal) {
+    sync::activate_sync_point("NOW", vec![sync::Action::EmitSignal(signal)], 1);
+    sync::on_sync_point("NOW").await.unwrap();
+}
+
+#[tokio::test]
+#[serial]
```

Comment:
Perhaps to avoid serial requirement, we can introduce the concept of a "session", somehow, rather than rely on global var in the form of lazy static.

Especially considering each test could be quite long (10s-100s?)

Open to ideas.

---

In summary of the other thread, we can use a thread-local random u64 session_id shared by all of the services in a given test which run on the same thread, and the test itself.

Reply:
To make it more performant, instead of a global variable, we should initialize the desired struct as an Arc, and then pass the Arc pointer to the thread-local variable to be referenced by all of the services in a test.

Reply:
In order for this to work with a multi-threaded runtime with > 1 thread, we could possibly use https://docs.rs/tokio/0.1.22/tokio/runtime/struct.Builder.html#method.after_start

Reply:
Anw, this can be done in a follow up PR, it is not a crucial blocker. 

In fact, I'd prefer this PR get merged more quickly so we can start adding other tests and iterating on the framework.

Reply:
Thanks for these suggestions. I agree let's iterate it later.